js/app.js

```diff
@@ -1,39 +1,115 @@
+  const result = parseNum1 + parseNum2;
+  console.log(`${parseNum1} + ${parseNum2} = ${result}`);
+
+  // 4. dodaj do historii operacji to działanie w fomie: 1 + 1 = 2
+  if (result) {
+    this.history.push(`${parseNum1} + ${parseNum2} = ${result}`);
+  }
+};
+
+Calculator.prototype.substract = function (num1, num2) {
+  const parseNum1 = parseFloat(num1);
+  const parseNum2 = parseFloat(num2);
+  console.log(typeof parseNum1, typeof parseNum2);
+
+  const result = parseNum1 - parseNum2;
+  console.log(`${parseNum1} - ${parseNum2} = ${result}`);
+
+  if (result) {
+    this.history.push(`${parseNum1} - ${parseNum2} = ${result}`);
+  }
+};
+
+Calculator.prototype.multiply = function (num1, num2) {
+  const parseNum1 = parseFloat(num1);
+  const parseNum2 = parseFloat(num2);
+  console.log(typeof parseNum1, typeof parseNum2);
+
+  const result = parseNum1 * parseNum2;
+  console.log(`${parseNum1} * ${parseNum2} = ${result}`);
+
+  if (result) {
+    this.history.push(`${parseNum1} * ${parseNum2} = ${result}`);
+  }
+};
+
+Calculator.prototype.divide = function (num1, num2) {
+  const parseNum1 = parseFloat(num1);
+  const parseNum2 = parseFloat(num2);
+  console.log(typeof parseNum1, typeof parseNum2);
+
+  const result = parseNum1 / parseNum2;
+  console.log(`${parseNum1} / ${parseNum2} = ${result}`);
+
+  if (result) {
+    this.history.push(`${parseNum1} / ${parseNum2} = ${result}`);
+  }
+};
+
+Calculator.prototype.exponent = function (num1, num2) {
+  let parseNum1 = parseFloat(num1);
+  let parseNum2 = parseFloat(num2);
+  console.log(typeof parseNum1, typeof parseNum2);
+
+  let result = (parseNum1 **= parseNum2);
+
+  const displayNum1 = num1;
+  const displayNum2 = num2;
+
+  console.log(`${displayNum1} ^ ${displayNum2} = ${result}`);
+
+  if (result) {
+    this.history.push(`${displayNum1} ^ ${displayNum2} = ${result}`);
+  }
+};
 
 const calc = new Calculator();
+
 let action, promptContent, isCorrectAction, number1, number2;
-do { 
-    promptContent = 'Podaj jaką operację chcesz wykonać (+, -, *, /, ^) i potwierdź. \n'; // \n - znak nowej linii
-    promptContent += 'Jeśli chcesz zrezygnować wciśnij Anuluj. \n';
-    promptContent += 'Lista poprzednich operacji: \n' + calc.getHistoryAsString();
-
-    action = prompt(promptContent);
-    isCorrectAction = calc.isCorrectAction(action);
-    if(isCorrectAction) {
-        number1 = prompt('Podaj liczbę nr 1');
-        number2 = prompt('Podaj liczbę nr 2');
-
-        if(action === '+') {
-            calc.add(number1, number2);
-        }
+do {
+  promptContent = 'Podaj jaką operację chcesz wykonać (+, -, *, /, ^) i potwierdź. \n'; // \n - znak nowej linii
+  promptContent += 'Jeśli chcesz zrezygnować wciśnij Anuluj. \n';
+  promptContent += 'Lista poprzednich operacji: \n' + calc.getHistoryAsString();
+
+  action = prompt(promptContent);
+
+  isCorrectAction = calc.isCorrectAction(action);
+  const operationFunc = this.actions[action].bind(calc);
```

Comment:
Tutaj powinno być `calc` zamiast `this`, ponieważ jesteśmy poza "konstruktorem" (później to będzie klasa).
Można też zamienić to na funkcję (metodę) tj. `getAction(action)` i zdefiniować ją w ten sposób:
```
Calculator.prototype.getAction = function(action) {
  return this.actions[action].bind(calc);
}

```

PS. To rozwiązanie powinno być wew. if, ponieważ musimy mieć pewność, że ona istnieje.